projects/plugins/jetpack/functions.global.php

```diff
@@ -407,6 +407,91 @@ function jetpack_is_file_supported_for_sideloading( $file ) {
 	return in_array( $type, $supported_mime_types, true );
 }
 
+/**
+ * Add support for PPSX and related file types
+ *
+ * @param array  $file_data Array of file data.
+ * @param string $file Full path to the file.
+ * @param string $filename Name of the file (may differ from $file).
+ * @param array  $mimes Array of supported MIME types.
+ *
+ * @return array $file_data File extension, MIME type, and corrected filename.
+ */
+function jetpack_allow_pptx_files( $file_data, $file, $filename, $mimes ) {
+	// Remove itself to avoid potential infinite loops.
+	remove_filter( 'wp_check_filetype_and_ext', 'jetpack_allow_pptx_files', 1000, 4 );
+	if ( ! file_exists( $file ) ) {
+		return $file_data;
+	}
+
+	// If the file does not exist, don't bother doing the checks.
+	if ( ! empty( $file_data['type'] ) ) {
+		$_invalid = array(
+			'ext'             => false,
+			'type'            => false,
+			'proper_filename' => false,
+		);
+		if ( 'asc' === strtolower( $file_data['ext'] ) ) {
+			return $_invalid;
+		}
+		return $file_data;
+	}
+
+	// Try to get the file type with the extension based check, bail if it fails.
+	$file_type = wp_check_filetype( $filename, $mimes );
+	if ( empty( $file_type['type'] ) ) {
+		return $file_data;
+	}
+
+	if ( 'asc' === strtolower( $file_type['ext'] ) ) {
+		if ( wpcom_check_gpg_valid_key( $file ) ) {
```

Comment:
`wpcom_check_gpg_valid_key` doesn't exist in Jetpack, so that's not going to be possible to use here, unless we add a `WPCOM` check before.